src/compiler/backend/riscv64/code-generator-riscv64.cc

```diff
@@ -325,7 +325,7 @@ void EmitWordLoadPoisoningIfNeeded(CodeGenerator* codegen,
     __ bin_instr(i.TempRegister(1), i.OutputRegister(0),                       \
                  Operand(i.InputRegister(2)));                                 \
     __ store_conditional(i.TempRegister(1), MemOperand(i.TempRegister(0), 0)); \
-    __ BranchShort(&binop, eq, i.TempRegister(1), Operand(zero_reg));          \
+    __ BranchShort(&binop, ne, i.TempRegister(1), Operand(zero_reg));          \
```

Comment:
Please add a comment before each of these macros to explain what should happen. The code difficult to follow.